lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -698,131 +670,98 @@ chainSyncWithBlocks
+
+    client :: ChainSyncClient blk (Tip blk) m Void
+    client = ChainSyncClient clientStIdle
       where
         -- Client in the state 'Idle'. We wait for requests / commands on an
         -- 'TQueue'. Commands start a chain of messages and state transitions
         -- before finally returning to 'Idle', waiting for the next command.
         clientStIdle
-            :: RequestNextStrategy m 'Z
-            -> m (P.ClientPipelinedStIdle 'Z ByronBlock (Tip ByronBlock) m Void)
-        clientStIdle strategy = atomically (readTQueue queue) >>= \case
+            :: m (ClientStIdle blk (Tip blk) m Void)
+        clientStIdle = atomically (readTQueue queue) >>= \case
             CmdFindIntersection points respond -> pure $
-                P.SendMsgFindIntersect points (clientStIntersect respond)
-            CmdNextBlocks respond ->
-                pure $ strategy respond
+                SendMsgFindIntersect points (clientStIntersect respond)
+
+            CmdNextBlocks respond -> pure $
+                SendMsgRequestNext
+                    (clientStNext ([], 1000) respond)
+                    (pure $ clientStNext ([], 1) respond)
 
         clientStIntersect
-            :: (Maybe (Point ByronBlock) -> m ())
-            -> P.ClientPipelinedStIntersect ByronBlock (Tip ByronBlock) m Void
-        clientStIntersect respond = P.ClientPipelinedStIntersect
-            { recvMsgIntersectFound = \intersection _tip -> do
-                respond (Just intersection)
-                clientStIdle oneByOne
-
-            , recvMsgIntersectNotFound = \_tip -> do
-                respond Nothing
-                clientStIdle oneByOne
+            :: (Maybe (Point blk) -> m ())
+            -> ClientStIntersect blk (Tip blk) m Void
+        clientStIntersect respond = ClientStIntersect
+            { recvMsgIntersectFound = \intersection _tip ->
+                ChainSyncClient $ do
+                    respond (Just intersection)
+                    clientStIdle
+
+            , recvMsgIntersectNotFound = \_tip ->
+                ChainSyncClient $ do
+                    respond Nothing
+                    clientStIdle
             }
 
-        -- Simple strategy that sends a request and waits for an answer.
-        oneByOne
-            :: RequestNextStrategy m 'Z
-        oneByOne respond = P.SendMsgRequestNext
-            (collectResponses respond [] Zero)
-            (pure $ collectResponses respond [] Zero)
-
-        -- We only pipeline requests when it's safe to pipeline them. Safe
-        -- is when all blocks are considered 'stable' and immutable, i.e.
-        -- when we land more than `k` blocks from the tip, so when we start
-        -- more than `k + maxInFlight` blocks from the tip
-        --
-        --     0                       tip - k           tip
-        --     |---------------------------|-------------->
-        --               pipelined            one by one
-        --
-        pipeline
```

Comment:
When I started pipelining wasn't on master. 

Should be simple to revert this deletion now.

Could also factor out the logic to a block-agnostic shared location… or maybe in the next PR.